File: app/src/main/java/empire/stark/firststep/common/activity/BaseActivity.java
package empire.stark.firststep.common.activity;

import android.app.Fragment;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;

import javax.inject.Inject;

import dagger.android.AndroidInjection;
import dagger.android.DispatchingAndroidInjector;
import dagger.android.HasDispatchingFragmentInjector;
import dagger.android.support.HasDispatchingSupportFragmentInjector;

/**
 * Created by YEN_MINH on 3/9/2017.
 */

public class BaseActivity extends AppCompatActivity implements HasDispatchingFragmentInjector,
        HasDispatchingSupportFragmentInjector {
    @Inject
    DispatchingAndroidInjector<android.app.Fragment> frameworkFragmentInjector;
    @Inject
    DispatchingAndroidInjector<android.support.v4.app.Fragment> fragmentSupportInjector;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        AndroidInjection.inject(this);
        super.onCreate(savedInstanceState);
    }

    @Override
    public DispatchingAndroidInjector<android.support.v4.app.Fragment> supportFragmentInjector() {
        return fragmentSupportInjector;
    }

    @Override
    public DispatchingAndroidInjector<Fragment> fragmentInjector() {
        return frameworkFragmentInjector;
    }
}

File: app/src/main/java/empire/stark/firststep/common/BaseActivity.kt
package empire.stark.firststep.common

import android.app.Activity
import android.app.Fragment
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import dagger.android.*
import dagger.android.support.HasSupportFragmentInjector
import javax.inject.Inject

/**
 * Created by YEN_MINH on 3/9/2017.
 */

open class BaseActivity : AppCompatActivity(), HasActivityInjector, HasFragmentInjector, HasSupportFragmentInjector {


    @Inject lateinit var dispatchingAndroidInjector: DispatchingAndroidInjector<Activity>
    @Inject lateinit var fragmentInjector: DispatchingAndroidInjector<Fragment>
    @Inject lateinit var fragmentSupportInjector: DispatchingAndroidInjector<android.support.v4.app.Fragment>


    override fun onCreate(savedInstanceState: Bundle?) {
        AndroidInjection.inject(this)
        super.onCreate(savedInstanceState)
    }

    override fun activityInjector(): AndroidInjector<Activity> {
        return dispatchingAndroidInjector
    }

    override fun fragmentInjector(): AndroidInjector<Fragment> {
        return fragmentInjector
    }

    override fun supportFragmentInjector(): AndroidInjector<android.support.v4.app.Fragment> {
        return fragmentSupportInjector
    }
}
